File: src/main/java/com/DavidM1A2/AfraidOfTheDark/common/event/register/FurnaceFuelRegister.java
package com.DavidM1A2.afraidofthedark.common.event.register;

import com.DavidM1A2.afraidofthedark.common.constants.ModBlocks;
import net.minecraft.block.Block;
import net.minecraftforge.event.furnace.FurnaceFuelBurnTimeEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

import java.util.HashMap;
import java.util.Map;

/**
 * Class that registers all furnace fuels
 */
public class FurnaceFuelRegister
{
    // Mapping of block -> burn time
    private static final Map<Block, Integer> ITEM_TO_BURN_TIME = new HashMap<Block, Integer>()
    {{
        put(ModBlocks.GRAVEWOOD_SAPLING, 100);
    }};

    /**
     * Called to make certain blocks furnace smeltable
     *
     * @param event The event containing the block/item smelting info
     */
    @SubscribeEvent
    public void onFurnaceFuelBurnTimeEvent(FurnaceFuelBurnTimeEvent event)
    {
        // If we know of the burn time for the item then set the burn time for the item based on the hashmap
        if (ITEM_TO_BURN_TIME.containsKey(Block.getBlockFromItem(event.getItemStack().getItem())))
        {
            event.setBurnTime(ITEM_TO_BURN_TIME.get(Block.getBlockFromItem(event.getItemStack().getItem())));
        }
    }
}

File: src/main/kotlin/com/davidm1a2/afraidofthedark/common/event/register/FurnaceFuelRegister.kt
package com.davidm1a2.afraidofthedark.common.event.register

import com.davidm1a2.afraidofthedark.common.constants.ModBlocks
import net.minecraft.item.Item
import net.minecraftforge.event.furnace.FurnaceFuelBurnTimeEvent
import net.minecraftforge.eventbus.api.SubscribeEvent

/**
 * Class that registers all furnace fuels
 */
class FurnaceFuelRegister {
    // Mapping of block -> burn time
    private lateinit var itemToBurnTime: Map<Item, Int>

    /**
     * Called to make certain blocks furnace smeltable
     *
     * @param event The event containing the block/item smelting info
     */
    @SubscribeEvent
    fun onFurnaceFuelBurnTimeEvent(event: FurnaceFuelBurnTimeEvent) {
        if (!::itemToBurnTime.isInitialized) {
            itemToBurnTime = mapOf(
                ModBlocks.GRAVEWOOD_SAPLING.asItem() to 100,
                ModBlocks.GRAVEWOOD_FENCE.asItem() to 300,
                ModBlocks.GRAVEWOOD_FENCE_GATE.asItem() to 300,
                ModBlocks.MANGROVE_SAPLING.asItem() to 100,
                ModBlocks.MANGROVE_FENCE.asItem() to 300,
                ModBlocks.MANGROVE_FENCE_GATE.asItem() to 300,
                ModBlocks.SACRED_MANGROVE_SAPLING.asItem() to 1000,
                ModBlocks.SACRED_MANGROVE_FENCE.asItem() to 3000,
                ModBlocks.SACRED_MANGROVE_FENCE_GATE.asItem() to 3000
            )
        }

        // If we know of the burn time for the item then set the burn time for the item based on the hashmap
        event.burnTime = itemToBurnTime[event.itemStack.item] ?: return
    }
}